src/machine/machine_rp2040.go

```diff
@@ -114,3 +114,8 @@ func init() {
 	UART0.Interrupt = interrupt.New(rp.IRQ_UART0_IRQ, _UART0.handleInterrupt)
 	UART1.Interrupt = interrupt.New(rp.IRQ_UART1_IRQ, _UART1.handleInterrupt)
 }
+
+// GetCoreNumber returns the core number the call was made from.
+func GetCoreNumber() uint32 {
```

Comment:
This is a new API, and should therefore have some thought how it should be designed. For example, the ESP32 also has two cores and may need an API like this.

My suggestion would be this:

```go
func CurrentCore() int
func CoreCount() int
```

* No `Get` prefix, as is usual in Go (similar to `CPUFrequency()`)
* Return an `int`, which is a more common data type (and is the same size as `uint32` on most microcontrollers)
* Name the function differently. Initially I was a bit confused and thought that `GetCoreCount` meant the total count, not the current core.
* As a bonus: add a function to get the total number of cores. Alternatively, this might be named `NumCores` - not sure what the best name is.